src/mpid/ch4/netmod/ofi/ofi_dynproc.c

```diff
@@ -88,6 +89,11 @@ int MPIDI_OFI_dynproc_insert_conn(fi_addr_t conn, int rank, int state, int *conn
     MPIDI_OFI_global.conn_mgr.conn_table[conn_id].rank = rank;
     MPIDI_OFI_global.conn_mgr.conn_table[conn_id].state = state;
 
+    if (state == MPIDI_OFI_DYNPROC_CONNECTED_PARENT) {
+        mpi_errno = dynproc_post_recv_disconnect(conn_id);
```

Comment:
I suppose this results in a performance penalty for regular send/recv since matching will always have to examine this posted recv. But this is the dynamic use-case, so probably OK for now, especially if we can figure out a fix in libfabric at some point.

Reply:
Only when the applications spawn new processes and it only affects the root process. In a typical "master-workers" application, the "master" is not doing much other than orchestrating workers, so I think the impact can be tolerated. 

For singleton application that spawns up the whole world, hopefully, it just needs to spawn once and a single pre-posted recv shouldn't affect much performance.